Mathlib/Algebra/Category/Grp/IsFinite.lean

```diff
@@ -0,0 +1,59 @@
+/-
+Copyright (c) 2025 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+import Mathlib.Algebra.Category.Grp.Abelian
+import Mathlib.Algebra.Category.Grp.EpiMono
+import Mathlib.Algebra.Homology.ShortComplex.Ab
+import Mathlib.CategoryTheory.Abelian.SerreClass.Basic
+import Mathlib.Data.Finite.Prod
+
+/-!
+# The Serre class of finite abelian groups
+
+In this file, we define `isFinite : ObjectProperty Ab` and show
+that it is a Serre class.
+
+-/
+
+universe u
+
+open CategoryTheory Limits ZeroObject
+
+namespace Ab
+
+/-- The Serre class of finite abelian groups in the category `Ab`. -/
+def isFinite : ObjectProperty Ab.{u} :=
+  fun M ↦ Finite M
+
+@[simp]
+lemma prop_isFinite_iff (M : Ab.{u}) : isFinite M ↔ Finite M := Iff.rfl
+
+instance : isFinite.{u}.IsSerreClass where
+  exists_zero := by
+    refine ⟨.of (ULift.{u} (Fin 1)), ?_, by rw [prop_isFinite_iff]; infer_instance⟩
+    rw [IsZero.iff_id_eq_zero]
+    ext x
+    fin_cases x
+    rfl
```

Comment:
If you add the import `Mathlib.Algebra.Category.Grp.Zero`, then this can become
```suggestion
  exists_zero := ⟨.of PUnit, isZero_of_subsingleton _, by rw [prop_isFinite_iff]; infer_instance⟩
```